### PR TITLE
Introduce favicons.xml (MD5 fingerprints for favicon.ico files)

### PR DESCRIPTION
**Pull request by hdm**
## Description

This adds a new favicon.ico MD5 fingerprint database to recog.

## Motivation and Context

These fingerprints can help identify systems when less reliable methods are available.

## How Has This Been Tested?

Tested against a huge internal dataset, also rspec, and recog_verify

## Types of changes

- New database, no impact to existing Recog users.

## Checklist:
- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

File: xml/favicons.xml [hunk 1]
```diff
@@ -0,0 +1,1696 @@
...
+  <fingerprint pattern="^cee18e28257988b40028043e65a6c2a3$">
+    <description>TeamCity</description>
+    <example>cee18e28257988b40028043e65a6c2a3</example>
+    <param pos="0" name="service.vendor" value="JetBrains"/>
+    <param pos="0" name="service.product" value="TeamCity"/>
+    <param pos="0" name="service.certainty" value="0.5"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:jetbrains:teamcity:-"/>
+  </fingerprint>
+
+  <fingerprint pattern="^e48c482f8f5a8e5a6249b21a39f911e7$"><description>Cockroach DB Console</description><example>e48c482f8f5a8e5a6249b21a39f911e7</example><param pos="0" name="service.vendor" value="Cockroach Labs"/><param pos="0" name="service.product" value="CockroachDB"/>&gt;
```
Comment (rkirk-r7) [inline xml/favicons.xml:365]: formatting

Comment (hdm) [reply xml/favicons.xml]: thanks! I wonder what did that, maybe a hanging > character and the being autoformatted by the cpe script

**Comment by hdm** [comment]
Is there anything else needed to merge this?

**Comment by tsellers-r7** [comment]
Merged, thanks for the contribution @hdm

**Pull request merged**